src/modules/CRUD/views/MapPlayground/components/LayersControl/LayersControl.js

```diff
@@ -59,19 +109,38 @@ export class LayersControl extends AbstractMapControl {
           position={Position.BOTTOM_LEFT}
           content={(
             <div className="radioGroup layerGroup__item">
-              {layers.map(({ title, id }) => {
-                const defaultChecked = map.getLayoutProperty(id, 'visibility') === 'visible';
+              {layers.map(({ title, id, empty = false }) => {
+                const defaultChecked = !empty && map.getLayoutProperty(id, 'visibility') === 'visible';
                 return (
                   <Checkbox
                     className="bgLayer-radio"
+                    defaultChecked={defaultChecked}
+                    disabled={empty}
                     key={id}
+                    onChange={this.onChange}
                     label={title}
                     value={id}
-                    onChange={onChange}
-                    defaultChecked={defaultChecked}
                   />
                 );
               })}
+
+              {relations
+                .sort((a, b) => a.order - b.order)
```

Comment:
Is it a good idea to put all this logic outside of render function?

Reply:
You're right, i refacto this logic by using a state to sanitize all the things